lib/spack/env/cc

```diff
@@ -295,6 +317,8 @@ while [ -n "$1" ]; do
                     die "-Wl,-rpath was not followed by -Wl,*"
                 fi
                 rp="${arg#-Wl,}"
+            elif [[ "$arg" = "$dtags_to_disable" ]] ; then
+                :  # We want to remove explicitly this flag
```

Comment:
I think we need to add this in the `-Xlinker` case as well as the `-Wl,` case. I think that should be fairly straightforward, but it's valid syntax to say `gcc -Xlinker --enable-new-dtags ...` and we'll need to catch that.

When checking this I also realized that the `-Xlinker,` case is an invalid option, we probably should not have handling for that. I'm pretty sure `-Xlinker` must be followed by a space.

Reply:
Should be done f1c52be